File: Backend/C#/SuperHero.Core/Service/ServiceResponse.cs
using System.Net;

namespace SuperHero.Core.Service
{
    public class ServiceResponse
    {
        public DateTime IssuedOn { get; set; } = DateTime.Now;
        public HttpStatusCode? Status { get; set; }
        public string StatusDescription { get; set; }

        public void SetOk() => Status = HttpStatusCode.OK;
        public void SetInternalServerError() => Status = HttpStatusCode.InternalServerError;
    }
}

File: Backend/JAVA/application/src/main/java/com/superhero/service/ServiceResponse.java
package com.superhero.service;

import java.time.LocalDateTime;

import org.springframework.http.HttpStatus;

public class ServiceResponse {

    public LocalDateTime IssuedOn;
    public HttpStatus Status;
    public String StatusDescription;

    public LocalDateTime getIssuedOn() {
        return IssuedOn;
    }

    public void setIssuedOn() {
        IssuedOn = LocalDateTime.now();
    }

    public HttpStatus getStatus() {
        return Status;
    }

    public void setStatus(HttpStatus status) {
        Status = status;
    }

    public String getStatusDescription() {
        return StatusDescription;
    }

    public void setStatusDescription(String statusDescription) {
        StatusDescription = statusDescription;
    }

    public void SetOk() {
        this.Status = HttpStatus.OK;
    }

    public void SetInternalServerError() {
        this.Status = HttpStatus.INTERNAL_SERVER_ERROR;
    }
}
